core/store/src/db/colddb.rs

```diff
@@ -90,32 +100,39 @@ impl<D: Database> ColdDB<D> {
     }
 }
 
-impl<D: Database> super::Database for ColdDB<D> {
+impl<D: Database> Database for ColdDB<D> {
+    // For hot columns read the data from the hot db. For cold columns try to
+    // read the data from the hot db and if it's missing read from the cold db.
     fn get_raw_bytes(&self, col: DBCol, key: &[u8]) -> std::io::Result<Option<DBSlice<'_>>> {
+        let hot_result = self.hot.get_raw_bytes(col, key)?;
+
         if Self::is_hot_column(col) {
-            return self.hot.get_raw_bytes(col, key);
+            return Ok(hot_result);
         }
+
+        if hot_result.is_some() {
```

Comment:
Maybe bypass DbVersion and BlockMisc columns somewhere around here?
I like reading blockchain data from hot first, but utility columns should be easily accessible for cold db without thinking about it.